app/queries/admin/gdpr_delete_requests_query.rb

```diff
@@ -0,0 +1,19 @@
+module Admin
+  class GDPRDeleteRequestsQuery
+    QUERY_CLAUSE = "users_gdpr_delete_requests.email ILIKE :search OR " \
+                   "users_gdpr_delete_requests.username ILIKE :search".freeze
+
+    def self.call(relation: ::GDPRDeleteRequest.all, options: {})
```

Comment:
A personal preference of mine is to try to use named parameters.  My thoughts being that it's a bit more legible.  The `options:` named parameter "hides" the true parameters (namely `:role` and `:search`).

```ruby
def self.call(relation:  ::GDPRDeleteRequest.all, role: nil, search: nil)
```

This then makes inline documentation and method signatures a bit more discoverable by our intellisense editor functions.

If you go that route, you wouldn't need line 7 below (you already have them).

And you could do some kwargs parameter passing (e.g. "double splat") in the controller:

```ruby
Admin::GDPRDeleteRequestsQuery.call(**params.permit(:search, :role))
```

The `**params.permit(:search, :role)` will create a kwarg-friendly hash `:search` and `:role` parameters and pass that new hash as the parameters for `.call`

Reply:
Thanks for this, @jeremyf. I've made these adjustments, however everything is now broken and I'm struggling to backtrack where I want wrong along the way making these changes. Does anything immediately stand out to you? When I make all of the changes you suggested, I get the following error:
```
ArgumentError at /admin/member_manager/gdpr_delete_requests
unknown keyword: "search"
```
However, I'm not sure _why_ `search` is undefined. I'm still poking and `pry`-ing in an attempt to find a working solution, but since I've been looking at this for long, thought that a second pair of eyes might be able to catch something faster than I.

Reply:
Dredging up some stuff out of Slack,

I'm confounded why the `**params.permit(:search)` didn't work, but with the tests you wrote I was able to confirm that `Admin::GDPRDeleteRequestsQuery.call(search: params[:search]).page(params[:page]).per(50)` works (after I restarted `yarn e2e`).